### PR TITLE
[DOC] Add detailed documentation to the validator concept.

### PR DESCRIPTION
**Pull request by smehringer**
Resolves https://github.com/seqan/product_backlog/issues/152

### CODE REVIEW

File: include/seqan3/argument_parser/validators.hpp [hunk 1]
```diff
@@ -40,6 +40,18 @@ namespace seqan3
  *
  * \details
  *
+ * When adding (positional) options to the seqan3::argument_parser you may pass also pass a
```
Comment (joergi-w) [inline include/seqan3/argument_parser/validators.hpp:43]:
```suggestion
 * When adding (positional) options to the seqan3::argument_parser you may pass a
```

**Review by joergi-w — CHANGES_REQUESTED** [review]

**Review by joergi-w — APPROVED** [review]

File: include/seqan3/argument_parser/validators.hpp [hunk 2]
```diff
@@ -40,6 +40,18 @@ namespace seqan3
  *
  * \details
  *
+ * When adding (positional) options to the seqan3::argument_parser you may pass a
+ * [functor](https://stackoverflow.com/questions/356950/what-are-c-functors-and-their-uses) that models
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp:44]:
```suggestion
 * [function object](https://en.cppreference.com/w/cpp/named_req/FunctionObject) that models
```

File: include/seqan3/argument_parser/validators.hpp [hunk 3]
```diff
@@ -40,6 +40,18 @@ namespace seqan3
...
+ * seqan3::validator which checks the option value provided by the user for some constraint.
+ *
+ * SeqAn provides several common-use-case validators, e.g. the seqan3::arithmetic_range_validator
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp:47]:
```suggestion
 * SeqAn provides several common-use-case validators, e.g. the seqan3::arithmetic_range_validator.
```

File: include/seqan3/argument_parser/validators.hpp [hunk 4]
```diff
@@ -40,6 +40,18 @@ namespace seqan3
...
+ *
+ * \include test/snippet/argument_parser/validators_2.cpp
+ *
+ * You can learn more about SeqAn validators in our tutorial \ref section_validation.
+ *
+ * See the concept requirements for detailed information if you would like to implement your own validator.
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp:53]:
```suggestion
 * To implement your own validator please refer to the detailed concept description below.
```
?

**Review by rrahn — CHANGES_REQUESTED** [review]
Some suggestions. Otherwise looks good

**Review by rrahn — APPROVED** [review]

**Pull request merged**